sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py

```diff
@@ -101,16 +105,63 @@ def __init__(self, name):
 
         controller_list_param = rospy.get_param("/move_group/controller_list")
 
+        robot_description = rospy.get_param("/robot_description")
+        self._joint_limits = dict()
+        self._read_joint_limits(robot_description)
+
         # create dictionary with name of controllers and corresponding joints
         self._controllers = {item["name"]: item["joints"] for item in controller_list_param}
 
         self._set_up_action_client(self._controllers)
 
+        self._set_points_lock = threading.Lock()
+        self._are_set_points_ready = False
+        self._set_points_cv = threading.Condition()
+        self._set_points = {}
+
+        self._set_up_set_points_subscribers(self._controllers)
+
         self.tf_buffer = tf2_ros.Buffer()
         self.listener = tf2_ros.TransformListener(self.tf_buffer)
 
         threading.Thread(None, rospy.spin)
 
+        self._wait_for_set_points()
+
+    def _read_joint_limits(self, robot_description):
+        robot_dom = minidom.parseString(robot_description)
+        robot = robot_dom.getElementsByTagName('robot')[0]
+
+        for child in robot.childNodes:
+            if child.nodeType is child.TEXT_NODE:
+                continue
+            if child.localName == 'joint':
+                jtype = child.getAttribute('type')
+                if jtype in ['fixed', 'floating', 'planar']:
+                    continue
+                name = child.getAttribute('name')
+
+                if jtype == 'continuous':
+                    minval = -pi
+                    maxval = pi
+                else:
```

Comment:
what is the joint_type passed in this else?

Reply:
It can be a revolute joint like this one https://github.com/shadow-robot/sr_teleop_hand_internal/blob/2ecab89fbb4fcddd929a07550885bfcdbdc2de83/sr_fingertip_hand_teleop/config/rh_fingers_description.xml#L157

Reply:
ok, thanks

can you then just move the name = child.getAttribute('name') before the first 'if' to slightly increase readability?

Reply:
the attribute name seems to be particular for nodes of type joint